service/collector/vpn_connection.go

```diff
@@ -0,0 +1,142 @@
+package collector
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-06-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sync/errgroup"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/azure-operator/client"
+	"github.com/giantswarm/azure-operator/service/controller/setting"
+)
+
+var (
+	vpnConnectionDesc *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName("azure_operator", "vpn_connection", "info"),
+		"VPN connection informations.",
+		[]string{
+			"id",
+			"name",
+			"location",
+			"connection_type",
+			"connection_status",
+			"provisioning_state",
+		},
+		nil,
+	)
+)
+
+type VPNConnectionConfig struct {
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+
+	AzureSetting             setting.Azure
+	HostAzureClientSetConfig client.AzureClientSetConfig
+}
+
+type VPNConnection struct {
+	k8sClient kubernetes.Interface
+	logger    micrologger.Logger
+
+	azureSetting             setting.Azure
+	hostAzureClientSetConfig client.AzureClientSetConfig
+}
+
+func NewVPNConnection(config VPNConnectionConfig) (*VPNConnection, error) {
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if err := config.AzureSetting.Validate(); err != nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.AzureSetting.%s", config, err)
+	}
+	if err := config.HostAzureClientSetConfig.Validate(); err != nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.HostAzureClientSetConfig.%s", config, err)
+	}
+
+	d := &VPNConnection{
```

Comment:
Here it is `d`. Below it is `r`. Neither is really reasonable IMO. Maybe choose a better one and use it everywhere. That would be really cool.